deploy/crds.yaml

```
@@ -513,121 +696,232 @@ spec:
                 type: object
               host:
+                description: The host (domain name) of the server. Must be a valid
+                  subdomain as defined in RFC 1123, such as my-app or hello.example.com.
+                  When using a wildcard domain like *.example.com the domain must
+                  be contained in double quotes. The host value needs to be unique
+                  among all Ingress and VirtualServer resources. See also Handling
+                  Host and Listener Collisions.
                 type: string
               ingressClassName:
+                description: Specifies which Ingress Controller must handle the VirtualServer
+                  resource.
                 type: string
               listener:
-                description: TransportServerListener defines a listener for a TransportServer.
+                description: Sets a custom HTTP and/or HTTPS listener. Valid fields
+                  are listener.http and listener.https. Each field must reference
+                  the name of a valid listener defined in a GlobalConfiguration resource
                 properties:
                   name:
+                    description: The name of a listener defined in a GlobalConfiguration
+                      resource.
                     type: string
                   protocol:
+                    description: The protocol of the listener.
                     type: string
                 type: object
               serverSnippets:
+                description: Sets a custom snippet in server context. Overrides the
+                  server-snippets ConfigMap key.
                 type: string
               sessionParameters:
-                description: SessionParameters defines session parameters.
+                description: The parameters of the session to be used for the Server
+                  context
                 properties:
                   timeout:
+                    description: The timeout between two successive read or write
+                      operations on client or proxied server connections. See proxy_timeout
+                      directive. The default is 10m.
                     type: string
                 type: object
               streamSnippets:
+                description: Sets a custom snippet in the stream context. Overrides
+                  the stream-snippets ConfigMap key.
                 type: string
               tls:
-                description: TransportServerTLS defines TransportServerTLS configuration
-                  for a TransportServer.
+                description: The TLS termination configuration.
                 properties:
                   secret:
                     type: string
                 type: object
               upstreamParameters:
-                description: UpstreamParameters defines parameters for an upstream.
+                description: The UpstreamParameters are set on stream context
```

Old description was bettter in my opinion

```suggestion
                description: UpstreamParameters defines parameters for an upstream.
```